### PR TITLE
Change ExtensionContainerScope to be single instance instead of static

### PR DESCRIPTION
**Pull request by generik0**

### CODE REVIEW

**Comment by generik0** [comment]
@jonorossi Spike for discussion, not ment to be merged, nor build

File: src/Castle.Windsor/MicroKernel/Lifestyle/ScopedLifestyleManager.cs [hunk 1]
```diff
@@ -65,7 +65,17 @@ private ILifetimeScope GetScope(CreationContext context)
 			{
 				throw new ObjectDisposedException("Scope was already disposed. This is most likely a bug in the calling code.");
 			}
-			var scope = localScope.GetScope(context);
+
+			ILifetimeScope scope;
+			if (accessor.HasKernelScoping)
+			{
+				var container = Kernel.Resolve<IScopeContainer>();
```
Comment (jonorossi) [inline src/Castle.Windsor/MicroKernel/Lifestyle/ScopedLifestyleManager.cs:72]: What is `IScopeContainer`?

File: src/Castle.Windsor/MicroKernel/Lifestyle/Scoped/IScopeAccessor.cs [hunk 2]
```diff
@@ -32,5 +32,9 @@ public interface IScopeAccessor : IDisposable
 		/// <param name = "context">Current creation context</param>
 		/// <exception cref = "T:System.InvalidOperationException"> Thrown when scope cache could not be accessed.</exception>
 		ILifetimeScope GetScope(CreationContext context);
+		ILifetimeScope GetScope(CreationContext context, IKernel kernel);
+		bool HasKernelScoping { get; }
```
Comment (jonorossi) [inline src/Castle.Windsor/MicroKernel/Lifestyle/Scoped/IScopeAccessor.cs:36]: This is obviously a breaking change to `IScopeAccessor`, implemented by both Windsor facilities and users. It also is counter to the goal of scopes within a container.

File: src/Castle.Windsor.Extensions.DependencyInjection/WindsorServiceProviderFactoryBase.cs [hunk 3]
```diff
@@ -56,6 +56,12 @@ protected virtual void SetRootContainer(IWindsorContainer container)
 		{
 			rootContainer = container;
 			AddSubSystemToContainer(rootContainer);
+			var rootScope = CreateRootScope();
+			
+			container.Register(Component
```
Comment (jonorossi) [inline src/Castle.Windsor.Extensions.DependencyInjection/WindsorServiceProviderFactoryBase.cs:61]: Registering an `ILifetimeScope` in the container is likely going to cause all sorts of pain.

**Review by jonorossi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**